src/nvim/normal.c

```diff
@@ -3612,6 +3626,108 @@ void scroll_redraw(int up, long count)
 }
 
 /*
+ * Return length of line "lnum" for horizontal scrolling.
+ */
+static colnr_T scroll_line_len(linenr_T lnum)
+{
+    char_u  *p;
+    colnr_T col;
+    int   w;
+
+    p = ml_get(lnum);
+    col = 0;
+    if (*p != NUL)
+      for (;;)
+      {
+          w = chartabsize(p, col);
+          mb_ptr_adv(p);
+          if (*p == NUL)    /* don't count the last character */
+            break;
+          col += w;
+      }
+    return col;
+}
+
+/*
+ * Find longest visible line number.  If this is not possible (or not desired,
+ * by setting 'h' in "guioptions") then the current line number is returned.
+ */
+static linenr_T gui_find_longest_lnum(void)
+{
+    linenr_T ret = 0;
+
+    /* Calculate maximum for horizontal scrollbar.  Check for reasonable
+     * line numbers, topline and botline can be invalid when displaying is
+     * postponed. */
+    if (curwin->w_topline <= curwin->w_cursor.lnum
+      && curwin->w_botline > curwin->w_cursor.lnum
```

Comment:
Wrong indentation. `&&` should be on the same column as the very first `curwin`.